util/util.c

```diff
@@ -83,6 +83,35 @@ uint32_t find_last_set_bit(uint32_t word)
 	return last_set_bit;
 }
 
+/**
+ * Locate the closest element in an array.
+ */
+uint32_t find_closest(int32_t val,
+		      const int32_t *array,
+		      uint32_t size)
+{
+	uint32_t diff = abs((abs(array[0]) - abs(val)));
```

Comment:
Is the abs needed for each element?
If you have array[0] = 1 and val = -1 => diff will be 0
I think 
```suggestion
	uint32_t diff = abs(array[0] - val));
```
is better. Same for below

Reply:
Thanks, if fixed that.